### PR TITLE
[BottomAppBar] Correct cut-out arc angle.

### PR DESCRIPTION
**Pull request by romoore**
The cut-out for the FAB was using incorrect angles resulting in
sub-pixel alignment of the Paths. This could be contributing to our
flaky screenshot tests which are off by a single digit (in the RGB
codes) and a few pixels. Even better, the bottom app bar is now level
across its entire width rather than dipping slightly toward the FAB.

|Alignment|Before|After|
|--|--|--|
|Leading|bab-lead-before|bab-lead-after|
|Center|bab-center-before|bab-center-after|
|Trailing|bab-trail-before|bab-trail-after|

Part of #4929

### CODE REVIEW

File: components/BottomAppBar/src/private/MDCBottomAppBarLayer.m [hunk 1]
```diff
@@ -41,28 +41,34 @@ - (CGPathRef)pathWithCutFromRect:(CGRect)rect
                  layoutDirection:(UIUserInterfaceLayoutDirection)layoutDirection {
   UIBezierPath *bottomBarPath = [UIBezierPath bezierPath];
 
+  static CGFloat kStartAngle;
+  static CGFloat kEndAngle;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    kEndAngle = (CGFloat)asin(kMDCBottomAppBarFloatingButtonPositionY /
+                             kMDCBottomAppBarFloatingButtonRadius);
```
[the remark this answers is not in the document]
Comment (romoore) [reply components/BottomAppBar/src/private/MDCBottomAppBarLayer.m]: That's fair. I probably deleted a space by accident.

File: components/BottomAppBar/src/private/MDCBottomAppBarLayer.m [hunk 2]
```diff
@@ -87,14 +93,14 @@ - (CGPathRef)pathWithCutFromRect:(CGRect)rect
         [self pathWithCutLeft:bottomBarPath
                          width:width
                         height:height
-                    startAngle:startAngle
-                      endAngle:endAngle];
+                    startAngle:kStartAngle
+                      endAngle:kEndAngle];
```
[the remark this answers is not in the document]
Comment (romoore) [reply components/BottomAppBar/src/private/MDCBottomAppBarLayer.m]: Done.

File: components/BottomAppBar/src/private/MDCBottomAppBarLayer.m [hunk 3]
```diff
@@ -41,28 +41,34 @@ - (CGPathRef)pathWithCutFromRect:(CGRect)rect
                  layoutDirection:(UIUserInterfaceLayoutDirection)layoutDirection {
   UIBezierPath *bottomBarPath = [UIBezierPath bezierPath];
 
+  static CGFloat kStartAngle;
```
Comment (jverkoey) [inline components/BottomAppBar/src/private/MDCBottomAppBarLayer.m:44]: Can we document the math/reasoning behind these angle calculations?

**Review by jverkoey — APPROVED** [review]
LGTM with a request for documentation.

File: components/BottomAppBar/src/private/MDCBottomAppBarLayer.m (hunk 3 above)
Comment (romoore) [reply components/BottomAppBar/src/private/MDCBottomAppBarLayer.m:44]: Done

**Comment by romoore** [comment]
**Original path**
```objc
<UIBezierPath: 0x6000002a21c0; <MoveTo {0, 38}>,
 <LineTo {157.24340558082187, 38.418180942629022}>,
 <CurveTo {197.91818094262902, 58.256594419178136} {162.99720803641367, 55.128436622156599} {181.20792526310146, 64.010396874769938}>,
 <CurveTo {217.75659441917813, 38.418180942629007} {207.23189457352964, 55.049625633786704} {214.54962563378669, 47.731894573529623}>,
 <LineTo {375, 38}>,
 <LineTo {375, 192}>,
 <LineTo {0, 192}>,
```

**Updated path**
Note that the first `LineTo` has the correct y-position (38).

```objc
<UIBezierPath: 0x6000002a0660; <MoveTo {0, 38}>,
 <LineTo {157.10263169285867, 38}>,
 <CurveTo {197.5, 58.397368307141328} {162.62547919116659, 54.788002951024033} {180.71199704897597, 63.92021580544926}>,
 <CurveTo {217.89736830714133, 38} {207.15184471240445, 55.222144513133834} {214.72214451313383, 47.651844712404454}>,
 <LineTo {375, 38}>,
 <LineTo {375, 192}>,
 <LineTo {0, 192}>
```

**Pull request merged**